src/platform_impl/macos/window.rs

```diff
@@ -543,14 +569,13 @@ impl UnownedWindow {
     }
 
     #[inline]
-    pub fn set_cursor_position(
-        &self,
-        cursor_position: LogicalPosition,
-    ) -> Result<(), ExternalError> {
+    pub fn set_cursor_position(&self, cursor_position: Position) -> Result<(), ExternalError> {
         let window_position = self.inner_position().unwrap();
+        let dpi_factor = self.hidpi_factor();
+        let logical_cursor_position = cursor_position.to_logical(dpi_factor);
         let point = appkit::CGPoint {
-            x: (cursor_position.x + window_position.x) as CGFloat,
-            y: (cursor_position.y + window_position.y) as CGFloat,
+            x: (logical_cursor_position.x + window_position.x) as CGFloat,
```

Comment:
Since `inner_position` is now returning a `PhysicalPosition` instead of a `LogicalPosition`, this is now adding together the logical position of the cursor and the physical position of the window. The window's position ought to be converted to a logical position also.